Project/Source/Components/UI/ComponentCanvas.cpp

```diff
@@ -57,13 +58,15 @@ float ComponentCanvas::GetScreenFactor() {
 
 void ComponentCanvas::RecalculateSizeAndScreenFactor() {
 	if (dirty) {
+		LOG("RECALCULATED");
 		if (App->userInterface->IsUsing2D()) {
 			size = App->renderer->GetViewportSize();
 		} else {
 			size = GetScreenReferenceSize() * SCENE_SCREEN_FACTOR;
 		}
 		float2 factor = size.Div(screenReferenceSize);
 		screenFactor = factor.x < factor.y ? factor.x : factor.y;
+		dirty = false;
```

Comment:
I think it's this line that introduces a new bug!! The bug is when you load a new scene, everything is downscaled until you resize. Either remove it or fix it

Reply:
This should be right. Maybe there is a problem where Invalidate is not being called correctly.

Reply:
The bug is caused by the following function not calling Invalidate():
```
void ComponentCanvas::SetScreenReferenceSize(float2 screenReferenceSize_) {
	screenReferenceSize = screenReferenceSize_;
        Invalidate();
}
```

Reply:
Done. Had to make some further adjustments (Modifying 2D internal now calls all required invalidates